### PR TITLE
Create module for generating overall lighty test coverage - master

### PR DESCRIPTION
**Pull request by PeterSuna**
Current coverage is generated for all modules separately. This works perfectly fine for unit tests, but not for integration tests. Jacoco can create report only for modules which are defined in dependencies. So for example, if IT have dependency only for example application which start all required models. Test coverage will be applied only for example application, but not for tested modules.
Solution can be creating a module with dependencies to all lighty.io modules with IT and testable code. This module will generate overall coverage from aggregating all jacoco files from separate modules.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -46,6 +46,7 @@ jobs:
           -Dsonar.projectKey=${{ env.SONAR_PROJECT_KEY }}
           -Dsonar.organization=${{ env.SONAR_ORGANIZATION }}
           -Dsonar.host.url=https://sonarcloud.io
+          -Dsonar.coverage.jacoco.xmlReportPaths=$(find "$(pwd)" -path '*jacoco-aggregate/jacoco.xml')
```
Comment (vainyksi) [inline .github/workflows/build.yml:49]: What about to specify exact path to "report-module" jacoco.xml?

File: lighty-tests/test-report-aggregator/pom.xml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+    <!-- For the correct result of lighty coverage, is necessary to add here dependencies with
+     IT tests or testable code from lighty.  -->
+
+    <dependencies>
+        <!--lighty applications -->
+        <dependency>
+            <groupId>io.lighty.applications.rnc</groupId>
+            <artifactId>lighty-rnc-app</artifactId>
+            <version>15.0.0-SNAPSHOT</version>
```
Comment (vainyksi) [inline lighty-tests/test-report-aggregator/pom.xml:28]: versions should be controlled via dependency management in BOM pom or sufficient parent's pom file

File: lighty-tests/pom.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    <properties>
+        <maven.deploy.skip>true</maven.deploy.skip>
+        <maven.install.skip>true</maven.install.skip>
+    </properties>
+
+    <modules>
+        <module>test-report-aggregator</module>
+    </modules>
+
+</project>
```
Comment (vainyksi) [inline lighty-tests/pom.xml:21]: remove this file
& move `test report aggregator` to root

**Review by vainyksi — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (PeterSuna) [reply .github/workflows/build.yml]: done

File: lighty-tests/test-report-aggregator/pom.xml (hunk 2 above)
Comment (PeterSuna) [reply lighty-tests/test-report-aggregator/pom.xml]: done

File: lighty-tests/pom.xml (hunk 3 above)
Comment (PeterSuna) [reply lighty-tests/pom.xml]: done

**Review by vainyksi — APPROVED** [review]

**Pull request merged**